### PR TITLE
Added parameter to the autocorr plot and code for partial autocorr plot

### PR DESCRIPTION
**Pull request by colonder**
Added additional parameters to the autocorrelation_plot to be able to plot a specified number of lags and additional code for plotting the partial autocorrelation plot together with p and q value.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] Additional argument to the autocorrelation_plot allowing to plot the desired number of lags, thus increase clarity and readibility of the plot. Additional function allowing to plot partial autocorrelation code depending on the statsmodels library.

### CODE REVIEW

**Comment by gfyoung** [comment]
Thanks for the PR!  You're going to also need to add a `whatsnew` entry (likely in `0.24.0`) as well as tests.

cc @TomAugspurger

File: pandas/plotting/_misc.py [hunk 1]
```diff
@@ -602,12 +602,15 @@ def lag_plot(series, lag=1, ax=None, **kwds):
     return ax
 
 
-def autocorrelation_plot(series, ax=None, **kwds):
-    """Autocorrelation plot for time series.
+def autocorrelation_plot(series, n_lags=None, ax=None, **kwds):
```
Comment (TomAugspurger) [inline pandas/plotting/_misc.py]: Move `n_lags` after `ax` to not break backwards compat.

File: pandas/plotting/_misc.py [hunk 2]
```diff
@@ -602,12 +602,15 @@ def lag_plot(series, lag=1, ax=None, **kwds):
...
+    """Autocorrelation plot for time series. Prints p value
```
Comment (TomAugspurger) [inline pandas/plotting/_misc.py]: See the contributing guide on docstrings. First line should be a one-line summary.

File: pandas/plotting/_misc.py [hunk 3]
```diff
@@ -602,12 +602,15 @@ def lag_plot(series, lag=1, ax=None, **kwds):
...
+    as the lag value where the ACF chart crosses the upper
+    confidence interval for the first time.
 
     Parameters:
     -----------
     series: Time series
+    n_lags: Number of lags to be plotted
```
Comment (TomAugspurger) [inline pandas/plotting/_misc.py]:
```
n_lags : int, optional
    Number of lags to be plotted. Length of the series by default.
```

File: pandas/plotting/_misc.py [hunk 4]
```diff
@@ -638,8 +641,47 @@ def r(h):
...
     ax.set_ylabel("Autocorrelation")
-    ax.plot(x, y, **kwds)
+    if n_samples:
+        ax.plot(x[:n_lags], y[:n_lags], **kwds)
+    else:
+        ax.plot(x, y, **kwds)
     if 'label' in kwds:
         ax.legend()
     ax.grid()
+    print("q value: %d" % (x[y < z99 / np.sqrt(n)][0]))
```
Comment (TomAugspurger) [inline pandas/plotting/_misc.py]: We don't want to print things.

File: pandas/plotting/_misc.py [hunk 5]
```diff
@@ -638,8 +641,47 @@ def r(h):
     ax.axhline(y=-z99 / np.sqrt(n), linestyle='--', color='grey')
     ax.set_xlabel("Lag")
     ax.set_ylabel("Autocorrelation")
-    ax.plot(x, y, **kwds)
+    if n_samples:
```
Comment (TomAugspurger) [inline pandas/plotting/_misc.py]: Where's `n_samples` come from?

File: pandas/plotting/_misc.py (hunk 3 above)
Comment (TomAugspurger) [reply pandas/plotting/_misc.py]: What if `n_lags` is larger than the length of the series?

**Review by TomAugspurger — COMMENTED** [review]
Thanks,

could you add the things @gfyoung asked for, and also post an image of the default `n_lags=None` and a shorter `n_lags`?

**Comment by colonder** [comment]
Oh, alright, I will do my best to correct all the things. Thanks, for all the comments and hints.

**Comment by jreback** [comment]
can you rebase and update

**Comment by jreback** [comment]
closing as stale. if you'd like to continue pls ping.

**Pull request closed without merging**